packages/mongodb-access-adapter/datagrip-access-adapter/src/main/kotlin/com/mongodb/jbplugin/accessadapter/datagrip/adapter/DataGripMongoDbDriver.kt

```diff
@@ -216,3 +229,20 @@ internal class DataGripMongoDbDriver(
  * @return
  */
 fun LocalDataSource.isMongoDbDataSource(): Boolean = this.databaseDriver?.id == "mongo" || this.databaseDriver == null
+
+/**
+ * Returns true if the provided local data source has at least one active connection
+ * attached to it.
+ *
+ * @return
+ */
+fun LocalDataSource.isConnected(): Boolean =
+    DatabaseConnectionManager
+        .getInstance()
+        .activeConnections
+        .any { connection ->
+            connection.connectionPoint.dataSource == dataSource &&
+                runCatching {
+                    !connection.remoteConnection.isClosed && connection.remoteConnection.isValid(5)
```

Comment:
magic constant? :)

Reply:
Oh, you are totally right. :shame: on me.

The fun fact is that this integer is not used for anything in the JetBrains MongoDB implementation of things, so I'll just put it in a constant and that's it haha.